### PR TITLE
Changes ZAS knockdown cooldowns to take into account the time knocked down. 

### PR DESCRIPTION
**Pull request by Reviire**
Currently the proc which knocks you down has a cooldown assigned to it, but it doesn't take into account the stun time, meaning any cooldown less than the stun time is completely useless. Fixes this so it's more intuitive. airflow_stun_cooldown values of above 0 will have an affect, rather than 0-50 doing the same thing.

AAAHH forgot to fix my bleeding edge again.

:cl:
 * tweak: ZAS Airflow knockdowns take knockdown time into account in regards to the cooldown.

### CODE REVIEW

**Comment by Probe1** [comment]
Your PRs are cute and hopeless.  Touching ZAS is the servers no no danger zone but I'm rooting for you.

File: code/ZAS/Airflow.dm [hunk 1]
```diff
@@ -78,7 +79,7 @@ atom/movable/GotoAirflowDest(n)
 
 	if(knockdown <= 0)
 		to_chat(src, "<span class='warning'>The sudden rush of air knocks you over!</span>")
-	SetKnockdown(rand(1,5))
```
Comment (ghost) [inline code/ZAS/Airflow.dm:81]: Jesus fucking christ it was random all along?

File: code/ZAS/Airflow.dm [hunk 2]
```diff
@@ -43,7 +43,7 @@ atom/movable/GotoAirflowDest(n)
 /mob/proc/airflow_stun()
 	if(isDead() || (flags & INVULNERABLE) || (status_flags & GODMODE))
 		return FALSE
-	if(world.time < last_airflow_stun + zas_settings.Get(/datum/ZAS_Setting/airflow_stun_cooldown))
+	if(world.time < (last_airflow_stun + zas_settings.Get(/datum/ZAS_Setting/airflow_stun_cooldown) + 50))
```
Comment (ghost) [inline code/ZAS/Airflow.dm:46]: magic 50

**Comment by ghost** [comment]
this PR absolutely requires the balance tag

File: code/ZAS/Airflow.dm (hunk 1 above)
Comment (SonixApache) [reply code/ZAS/Airflow.dm:81]: 😂

**Comment by SonixApache** [comment]
ganbare

**Comment by Probe1** [comment]
@DrCelt why?

**Comment by Accipitermaris** [comment]
absolute madman

**Comment by Reviire** [comment]
@DrCelt Why? The balance is in the config. Could just set it to 0 if they want no delay between stuns.

File: code/ZAS/Airflow.dm (hunk 2 above)
Comment (DamianX) [reply code/ZAS/Airflow.dm:46]: Seconded, explain the meaning of this 50. Explain it in the code.

**Comment by ghost** [comment]
This is a change that affects balance; hence it needs a balance change tag

**Comment by ghost** [comment]
this knocks you down, if you're knocked down you're not balanced QED

File: code/ZAS/Airflow.dm (hunk 2 above)
Comment (jknpj) [reply code/ZAS/Airflow.dm:46]: EXPLAIN THIS MADNESS SO WE CAN END THE REVIEW

Comment (Reviire) [reply code/ZAS/Airflow.dm:46]: It's the knockdown. I didn't change the 5 so I just added a 50. To account for the knockdown time. A bit further down there's a "SetKnockdown(5)" which I didn't change.

**Comment by ShiftyRail** [comment]
This just needed some change to a define and was at +8
Sad

**Comment by jknpj** [comment]
Alright seems like no one is going to fix this

**Comment by Exxion** [comment]
It also had an issue with variable scope but whatever

EDIT (5 months later): What did I mean by this?

**Pull request closed without merging**